src/point-solutions/libraries/src/token-vending-machine/src/token-vending-machine.ts

```diff
@@ -0,0 +1,171 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { STSClient, AssumeRoleCommand } from "@aws-sdk/client-sts";
+import * as jwt from "jsonwebtoken";
+import { JwksClient } from "jwks-rsa";
+/**
+ * The Token Vending Machine library offers a solution for dynamically assuming an
+ * ABAC role and obtaining tenant-scoped credentials. It conceals the intricacies
+ * of managing and generating these scoped credentials. By utilizing these scoped credentials,
+ * tenant isolation is enforced when accessing tenant-specific resources.
+ */
+export class TokenVendingMachine {
+  private sts: STSClient;
+
+  constructor(private isValidateToken: boolean = false) {
+    this.sts = new STSClient();
+  }
+
+  private async validateJwt(
+    jwtToken: string,
+    idpDetails: string,
+  ): Promise<boolean> {
+    const idpDetailsJson = JSON.parse(idpDetails);
+    const issuer = idpDetailsJson.issuer;
+    const audience = idpDetailsJson.audience;
+
+    const client = new JwksClient({
+      jwksUri: `${issuer}.well-known/jwks.json`,
+    });
+
+    const decodedToken = jwt.decode(jwtToken, { complete: true });
+    if (!decodedToken) {
+      console.error("Decoding token has failed");
+      return false;
+    }
+
+    const kid = decodedToken.header.kid;
+    const key = await client.getSigningKey(kid);
+    const signingKey = key.getPublicKey();
+
+    try {
+      jwt.verify(jwtToken, signingKey, {
```

Comment:
This method returns the payload decoded if the signature is valid and optional expiration, audience, or issuer are valid. If not, it will throw the error.